### PR TITLE
RSDK-6865 android ML and streaming libraries

### PR DESCRIPTION
**Pull request by abe-winter**
## What changed
- move android build stuff to its own makefile, `android.make`, with `.aar` builds for the 2 platforms we target
- add android-specific build instructions for tflite and x264
- new droid.go entrypoint for gomobile
- build tag changes for: microphone, tflite, streaming. Despite the extra tagging, normal builds (untagged on linux) should not change. As you read through please call out any place you notice that *does* change the untagged build.
- ForceRestart flag in web/server for triggering needsRestart programmatically (used by droid GUI via droid.go)

Note: droid.yml is failing on this branch because it's using the workflow definitions from `main`. Once this PR is merged, droid.yml will be disabled by the `if: false` change in this branch.

## Why
- Android ML and streaming support

## Follow-up changes
Once this is merged, follow-on changes will come in smaller PRs:
- edit android CI to build the new .aar instead of the old go binary
- home directory and module stuff that is particularly strange on android for OS reasons + needs a design conversation

### CODE REVIEW

File: etc/android/build-tflite.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  pushd ~/tensorflow/build_$arch
+  cmake -DCMAKE_TOOLCHAIN_FILE=${ANDROID_NDK}/build/cmake/android.toolchain.cmake \
+  -DANDROID_ABI=$arch ../tensorflow-2.12.0/tensorflow/lite/c
+  cmake --build . -j
+  ${NDK_TOOLCHAIN}/bin/llvm-strip --strip-unneeded libtensorflowlite_c.so
+  local dest=${ROOT_DIR}/services/mlmodel/tflitecpu/android/jni/$arch
+  mkdir -p $dest
+  cp libtensorflowlite_c.so $dest
+  popd
+}
```
Comment (bhaney) [inline etc/android/build-tflite.sh:28]: just wanted to say, wow, it's amazing how difficult it is to get a shared library for tflite

File: etc/android/tflite.patch
Comment (bhaney) [inline etc/android/tflite.patch:1]: how did you realize you needed a tflite patch?

**Review by bhaney — APPROVED** [review]
Nice work, this looks like it was tedious to do

File: etc/android/tflite.patch
Comment (abe-winter) [reply etc/android/tflite.patch:1]: the normal way (i.e. edaniels submitted a PR with the fix)

**Pull request merged**